lib/improver/blending/weighted_blend.py

```diff
@@ -192,24 +157,27 @@ def process(self, cubes_in, cycletime=None):
         cubes_in = (
             [cubes_in] if isinstance(cubes_in, iris.cube.Cube) else cubes_in)
 
-        # if input is already a single cube, return with updated metadata
-        if len(cubes_in) == 1:
-            if "model" in self.blend_coord:
-                cubes_in[0].attributes[self.model_id_attr] = "blend"
-            return cubes_in[0]
+        # copy cubes to avoid modifying inputs
+        if ("model" in self.blend_coord and
```

Comment:
This is a bit flexible partly because what's passed into the CLI in our suite and tests is actually "model_configuration" (the string coordinate) rather than "model_id" (the numerical coordinate).  On a higher level, since neither of these coordinates actually exist on the input cube (but are created and then destroyed as part of the blending process), it's nice to allow the user to specify "model" as the blending coordinate from the CLI without worrying about the internals of the plugin.  I would envisage this as being the easiest interface for a general user, to expect the string "model" input to the CLI and handle the logic of that internally to the plugin, as we do here.

Reply:
On the other hand this may be dealt with further up.  I'll check.